extension/getSelection.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * Check if the selection object is empty
+ * @param {Object} selection The selected page
+ * @returns {boolean}
+ */
+function isEmpty(selection) {
+  if (Object.keys(selection).length === 0) {
+    return true;
+  }
+  return false;
+}
+
+module.exports = async (context, { selection }) => {
+  try {
+    let updatedSelection = {};
+    let cachedSelection;
+
+    // get selection from cache
+    try {
+      cachedSelection = await context.storage.extension.get('selection');
```

Comment:
as i already mentioned at one of the frontend files, i think you should use two pipelines - one for fetching the current selection, one for updating it. 

additionally please don't store the value inside the `extension` storage, but use at least the `device` and maybe also the `user` storage. the extension storage is a global storage for all users. so when one user sets a selection, this will be the default one for users who restart their apps.

Reply:
done